### PR TITLE
Deprecate and replace PayloadInterface status constants

### PR DESCRIPTION
**Pull request by elazar**
I limited the codes included to the ones we're likely to use most often. Feel free to suggest others. Here's [a more comprehensive list of them](https://github.com/lukasoppermann/http-status/blob/master/src/Httpstatuscodes.php).

### CODE REVIEW

File: src/PayloadInterface.php [hunk 1]
```diff
@@ -6,25 +6,85 @@
...
      */
     const UNKNOWN = 40;
 
     /**
+     * @var integer
+     */
+    const STATUS_OK = 'OK';
+
+    /**
+     * @var integer
```
Comment (ameech) [inline src/PayloadInterface.php]: Maybe I'm confused, but this doesn't look like an integer.

File: src/PayloadInterface.php [hunk 2]
```diff
@@ -6,25 +6,85 @@
 {
     /**
      * @var integer All good.
+     * @deprecated 1.1.0
+     * @see \Equip\Adr\PayloadInterface::STATUS_OK
```
Comment (shadowhand) [inline src/PayloadInterface.php:10]: :+1:

File: src/PayloadInterface.php (hunk 1 above)
Comment (shadowhand) [inline src/PayloadInterface.php]: :warning:

File: src/PayloadInterface.php [hunk 3]
```diff
@@ -6,25 +6,85 @@
...
+
+    /**
+     * @var integer
+     */
+    const STATUS_FORBIDDEN = 'Forbidden';
+
+    /**
+     * @var integer
+     */
+    const STATUS_NOT_FOUND = 'Not Found';
```
Comment (shadowhand) [inline src/PayloadInterface.php:85]: Despite what I said before, we should add all of the statuses that are found in http-status package. That's safest and will reduce the need to version up later on.

File: src/PayloadInterface.php [hunk 4]
```diff
@@ -6,25 +6,85 @@
...
+
+    /**
      * Create a copy of the payload with the status.
      *
      * @param int $status
```
Comment (shadowhand) [inline src/PayloadInterface.php]: `int|string`

Comment (shadowhand) [inline src/PayloadInterface.php]: Same for `@return` in `getStatus`.

Comment (elazar) [inline src/PayloadInterface.php]: Why either or if we won't be using `int`? The translation from `string` to `int` shouldn't happen until the response object with the `string` value reaches the formatter.

File: src/PayloadInterface.php (hunk 3 above)
Comment (elazar) [inline src/PayloadInterface.php:85]: Not sure I follow your reasoning.

There's nothing to stop someone from using a status other than what's in `PayloadInterface`; the library can translate any valid status given to it.

It seems doubtful that we'll suddenly find a lot of statuses we actually need, so the need to version up should be few and far between.

I'd rather not clutter up the file with status codes we're unlikely to ever actually use (e.g. 418).

File: src/PayloadInterface.php [hunk 5]
```diff
@@ -6,28 +6,88 @@
...
+
+    /**
+     * @var string
+     */
+    const STATUS_FORBIDDEN = 'Forbidden';
+
+    /**
+     * @var string
+     */
+    const STATUS_NOT_FOUND = 'Not Found';
```
Comment (shadowhand) [inline src/PayloadInterface.php:85]: What about adding all the possible statuses? Eventually we'll need something that hasn't been added and then we're jumping through versioning. If we add them all now we can avoid that.

Comment (elazar) [inline src/PayloadInterface.php:85]: Missed [this comment](https://github.com/equip/adr/pull/2#discussion_r52638858) ?

File: src/PayloadInterface.php (hunk 3 above)
Comment (shadowhand) [inline src/PayloadInterface.php:85]: Missed this before. Good enough for now.

**Comment by shadowhand** [comment]
:+1: 

[Approved with PullApprove](https://pullapprove.com/equip/adr/pull-request/2/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=equip/adr)

**Pull request merged**